gradient_boosting.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+from sklearn.base import RegressorMixin
+from tree import DecisionTreeRegressor
+
+
+class BaseGradientBoostingRegressor:
+
+    def __init__(self,
+                 n_estimators: 5,
```

Comment:
**This is annotation, but not the default parameter**. Default parameter can be set using '=' (for example `n_estimators=100`. Please, read about [annotation](https://www.python.org/dev/peps/pep-3107/) in python and module [typing](https://docs.python.org/3/library/typing.html).